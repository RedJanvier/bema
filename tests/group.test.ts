import './__setup'
import { noop } from 'lodash'
import { stripKeys } from './__helpers'

describe('afterAll', () => {
  test.todo('receives provider context from the group+ancestors but not other branches or downstrema')
  test('afterAll is called once after all cases in group have run', async () => {
    let ga = bema.parameter('name').group('a').afterAll(f1)
    ga.case({ name: '1' }).run(noop)
    ga.case({ name: '2' }).run(noop)
    ga.case({ name: '3' }).run(noop)
    await run(bema)
    expect(f1.mock.calls.length).toEqual(1)
    expect(f1.mock.calls[0].length).toEqual(1)
    expect(stripKeys(['stats', 'time'], f1.mock.calls[0][0])).toMatchSnapshot()
  })
})

describe('afterEach', () => {
  test('is called after each case', async () => {
    const ga = bema.group('a').afterEach(f1)
    ga.case().run(noop)
    ga.case().run(noop)
    ga.case().run(noop)
    await run(bema)
    expect(f1.mock.calls.length).toEqual(3)
  })
})
