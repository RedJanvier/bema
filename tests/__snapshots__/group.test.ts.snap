// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`afterAll afterAll is called once after all cases in group have run 1`] = `
Object {
  "$info": Object {
    "caseResults": Object {
      "all": Array [
        Object {
          "report": Object {
            "group": "a",
            "matrix": Array [],
            "name": "name:1",
            "parameters": Object {
              "name": "1",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "1",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
          "returnValue": undefined,
        },
        Object {
          "report": Object {
            "group": "a",
            "matrix": Array [],
            "name": "name:2",
            "parameters": Object {
              "name": "2",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "2",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
          "returnValue": undefined,
        },
        Object {
          "report": Object {
            "group": "a",
            "matrix": Array [],
            "name": "name:3",
            "parameters": Object {
              "name": "3",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "3",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
          "returnValue": undefined,
        },
      ],
      "byMatrix": Object {
        "": Array [
          Object {
            "report": Object {
              "group": "a",
              "matrix": Array [],
              "name": "name:1",
              "parameters": Object {
                "name": "1",
              },
              "parametersFromMatrix": Object {},
              "parametersFromNotMatrix": Object {
                "name": "1",
              },
              "quickMode": true,
              "stats": "STRIPPED",
            },
            "returnValue": undefined,
          },
          Object {
            "report": Object {
              "group": "a",
              "matrix": Array [],
              "name": "name:2",
              "parameters": Object {
                "name": "2",
              },
              "parametersFromMatrix": Object {},
              "parametersFromNotMatrix": Object {
                "name": "2",
              },
              "quickMode": true,
              "stats": "STRIPPED",
            },
            "returnValue": undefined,
          },
          Object {
            "report": Object {
              "group": "a",
              "matrix": Array [],
              "name": "name:3",
              "parameters": Object {
                "name": "3",
              },
              "parametersFromMatrix": Object {},
              "parametersFromNotMatrix": Object {
                "name": "3",
              },
              "quickMode": true,
              "stats": "STRIPPED",
            },
            "returnValue": undefined,
          },
        ],
      },
    },
    "cases": Array [
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:1",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "1",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "1",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:2",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "2",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "2",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:3",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "3",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "3",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
    ],
    "casesSelected": Array [
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:1",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "1",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "1",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:2",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "2",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "2",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
      Object {
        "afterCallbacks": Array [],
        "beforeCallbacks": Array [],
        "matrixChain": Array [],
        "name": "name:3",
        "parameterDefinitions": Array [
          Object {
            "0": "n",
            "1": "a",
            "2": "m",
            "3": "e",
          },
        ],
        "parameters": Object {
          "name": "3",
        },
        "parametersFromMatrix": Object {},
        "parametersFromNotMatrix": Object {
          "name": "3",
        },
        "providerStack": Array [],
        "runCallback": [Function],
      },
    ],
    "name": "a",
    "time": "STRIPPED",
  },
}
`;
